ddtrace/ddtrace.go

```diff
@@ -122,6 +125,9 @@ type StartSpanConfig struct {
 	// Force-set the SpanID, rather than use a random number. If no Parent SpanContext is present,
 	// then this will also set the TraceID to the same value.
 	SpanID uint64
+
+	// Context holds the context associated with this span.
+	Context context.Context
```

Comment:
I was trying to avoid adding a public field to this struct, but I couldn't figure out a way to avoid it. If anybody has any ideas, please let me know!

Reply:
This is nasty, but I have an idea. What if we do this to `StartSpan`?
```go
func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOption) ddtrace.Span {
    return startSpanFromContext(gocontext.Background(), operationName, options...)
}
```
Then, you can use `startSpanFromContext` inside `StartSpanFromContext` and drop the `withContext` option along with this field.

Would that work?

Reply:
Ah, yes. True. Damn. Definitely not better with type assertions. Can we at least make it usable then? By that I mean, can we check if there is a span inside of it, and if yes, use it as a parent span? What do you think of this idea? It would be something like this:
```diff
diff --git a/ddtrace/tracer/tracer.go b/ddtrace/tracer/tracer.go
index 06cc12f..b1cf53d 100644
--- a/ddtrace/tracer/tracer.go
+++ b/ddtrace/tracer/tracer.go
@@ -378,6 +378,12 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 		if ctx, ok := opts.Parent.(*spanContext); ok {
 			context = ctx
 		}
+	} else if opts.Context != nil {
+		if p, ok := SpanFromContext(opts.Context); ok {
+			if ctx, ok := p.Context.(*spanContext); ok {
+				context = ctx
+			}
+		}
 	}
 	id := opts.SpanID
 	if id == 0 {

```